fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -3254,6 +3254,16 @@ public void addPartition(Database db, String tableName, AddPartitionClause addPa
         DataProperty dataProperty = singlePartitionDesc.getPartitionDataProperty();
         Preconditions.checkNotNull(dataProperty);
 
+        // check replica quota if this operation done
```

Comment:
The check here, and check at line L3851-3858 and L3895-3907 can be done in `Catalog.createPartitionWithIndices()`, before `createTablets()`

https://github.com/apache/incubator-doris/blob/4c6cbdf4635e48a3465bdfd7302cd3a0faabba42/fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java#L3527-L3537

Reply:
L3895-3907,when there are multi partitions, I don't think we can check quota in single partition.